src/PantheonTerminalWindow.vala

```diff
@@ -658,6 +670,8 @@ namespace PantheonTerminal {
             restorable_terminals.insert (term.terminal_id, term);
             tab.restore_data = term.terminal_id;
 
+            ((Gtk.Container)tab.page).remove(term);
```

Comment:
Sorry, I don't know what you meant. If after this line add `term.destroy ()`, then all term signals will be disabled